File: app/src/main/java/me/ibore/libs/demo/MainActivity.java
package me.ibore.libs.demo;

import android.os.Bundle;

import androidx.annotation.Nullable;

import me.ibore.libs.basic.XActivity;
import me.ibore.libs.demo.fm.HomeFragment;
import me.ibore.libs.rxbus.Subscribe;
import me.ibore.libs.rxbus.ThreadMode;
import me.ibore.libs.util.BarUtils;
import me.ibore.libs.util.LogUtils;
import me.ibore.libs.util.ToastUtils;

public class MainActivity extends XActivity {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void onBindView(@Nullable Bundle savedInstanceState) {
        BarUtils.addMarginTopEqualStatusBarHeight(rootView);

    }

    @Override
    protected void onBindData() {
        HomeFragment homeFragment = new HomeFragment();
        homeFragment.onDestroy();
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment, new HomeFragment())
                .commit();


    }


    @Subscribe(code = 11, threadMode = ThreadMode.MAIN)
    public void ss() {
        ToastUtils.showShort("dddddddddd");
        LogUtils.d("wwwwwwwwwwwwwww");
    }

    @Subscribe(code = 22, threadMode = ThreadMode.MAIN)
    public void ss(Test test) {
        ToastUtils.showShort("dfffffffffffffffff");
        LogUtils.d("fffffffffffffffffff");
    }


    @Subscribe(threadMode = ThreadMode.MAIN)
    public void sssss(String test) {
        ToastUtils.showShort(test);
        LogUtils.d(test);
    }
}

File: app/src/main/java/me/ibore/libs/demo/MainActivity.kt
package me.ibore.libs.demo

import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import me.ibore.libs.constant.PermissionConstants
import me.ibore.libs.util.PermissionUtils
import me.ibore.libs.util.ToastUtils

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        PermissionUtils
                .permission(PermissionConstants.STORAGE)
                .callback(object : PermissionUtils.Callback {
                    override fun onGranted(permissionsGranted: MutableList<String>?) {
                        ToastUtils.showShort("权限通过")
                    }

                    override fun onDenied(permissionsDeniedForever: MutableList<String>?, permissionsDenied: MutableList<String>?) {
                        AlertDialog.Builder(this@MainActivity)
                                .setMessage("dddddddddd")
                                .setPositiveButton("确认") { dialog, _ ->
                                    dialog.dismiss()
                                    PermissionUtils.openAppSettings()
                                }
                                .setNegativeButton("取消") { dialog, _ -> dialog.dismiss() }
                                .create().show()
                    }
                })
                .request()

    }
}